sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,258 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+   * causing the changes.
+   * @param applicationId application id.
+   * @returns a new student restriction object, that need to be saved.
+   */
+  async getStudentRestrictions(
+    scholasticStandingData: ScholasticStanding,
+    offeringIntensity: OfferingIntensity,
+    studentId: number,
+    auditUserId: number,
+    applicationId: number,
+  ): Promise<StudentRestriction> {
+    if (offeringIntensity === OfferingIntensity.fullTime) {
+      return this.getFullTimeStudentRestrictions(
+        scholasticStandingData,
+        studentId,
+        auditUserId,
+        applicationId,
+      );
+    }
+    if (offeringIntensity === OfferingIntensity.partTime) {
+      if (
+        scholasticStandingData.scholasticStanding ===
+          SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM ||
+        scholasticStandingData.scholasticStanding ===
+          SCHOLASTIC_STANDING_STUDENT_WITHDREW_FROM_PROGRAM
+      ) {
+        return this.studentRestrictionService.createRestrictionToSave(
+          studentId,
+          RestrictionCode.PTSSR,
+          auditUserId,
+          applicationId,
+        );
+      }
+    }
+  }
+
+  /**
+   * Get full time related restrictions for scholastic standing.
+   * When institution report withdrawal for a FT course application,
+   * add WTHD restriction to student.
+   * When institution report Withdrawal for a FT course on a student WITH a WTHD
+   * restriction add SSR restriction.
+   * When institution reports a change related to a FT application for unsuccessful
+   * completion and the total number of unsuccessful weeks hits minimum 68, add SSR
+   * restriction.
+   * If a ministry user resolves the SSR or WTHD restriction, and new withdrawal
+   * is reported, re add the above restrictions.
+   * If a ministry user resolves the SSR restriction, and new unsuccessful completion
+   * is reported, add the restriction (minimum is still at least 68).
+   * @param scholasticStandingData scholastic standing data.
+   * @param studentId student id.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   * @param applicationId application id.
+   * @returns a new student restriction object, that need to be saved.
+   */
+  async getFullTimeStudentRestrictions(
+    scholasticStandingData: ScholasticStanding,
+    studentId: number,
+    auditUserId: number,
+    applicationId: number,
+  ): Promise<StudentRestriction> {
+    if (
+      scholasticStandingData.scholasticStanding ===
+      SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM
+    ) {
+      if (!scholasticStandingData.numberOfUnsuccessfulWeeks) {
+        throw new Error(`number of unsuccessful weeks is empty.`);
+      }
+      const totalExistingUnsuccessfulWeeks =
+        await this.getTotalFullTimeUnsuccessfulWeeks(studentId);
+
+      // When total number of unsuccessful weeks hits minimum 68, add SSR restriction.
+      if (
```

Comment:
Are we checking anywhere if the 68 weeks restriction is not already in place?

Reply:
I searched for `68` in API, I didn't find any expected results, 
![image](https://user-images.githubusercontent.com/77353155/172732776-3197a7ac-ddf6-4769-a5f8-7486b152e170.png)

Reply:
It is not what I meant. Before creating an SSR restriction because the 68 weeks is passed, should we not check on DB if the student already has the restriction?

Reply:
As we talked about in the call, the ticket is updated and the expectation is clear now to me 😉